### PR TITLE
Added C++ implementation of Regula Falsi method

### LINKED ISSUE #1916: Regula_Falsi_Method in C++ implementation
I would like to implement Regula_Falsi_Method in C++14.Please assign this issue to me.

### PR DESCRIPTION
**Pull request by Nikhil-Patro**
Fixes #1916 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added a C++ implementation of Regula Falsi method to find roots of polynomials.

#### Languages Used:
- C++
#### Files Added:
- regula_falsi.cpp
> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by Nikhil-Patro** [comment]
Please review @asha15  @HardevKhandhar

File: Regula_Falsi_Method/regula_falsi.cpp [hunk 1]
```diff
@@ -0,0 +1,71 @@
+/* Girlscript Summer Of Code
+Regula Falsi Method in c
```
Comment (asha15) [inline Regula_Falsi_Method/regula_falsi.cpp]: this should be `Regula Falsi Method in c`

File: Regula_Falsi_Method/regula_falsi.cpp [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+In mathematics, the false position method or regula falsi is a very old method for solving an equation in one unknown,
+that, in modified form, is still in use.Regula Falsi method or the method of false position is a numerical method for 
+solving an equation in one unknown. It is quite similar to bisection method algorithm and is one of the oldest approaches.
+It was developed because the bisection method converges at a fairly slow speed. In simple terms, the method is the
+trial and error technique of using test ("false") values for the variable and then adjusting the test value according
+to the outcome.
+In simple terms, the method is the trial and error technique of using test ("false") values for the variable and
+then adjusting the test value according to the outcome. This is sometimes also referred to as "guess and check".
+Versions of the method predate the advent of algebra and the use of equations.*/
```
Comment (asha15) [inline Regula_Falsi_Method/regula_falsi.cpp]: one line should contain maximum 80 characters, break this line according to that
I feel  it's better if you make the commend short

File: Regula_Falsi_Method/regula_falsi.cpp [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
```
Comment (asha15) [inline Regula_Falsi_Method/regula_falsi.cpp]: remove this extra line

File: Regula_Falsi_Method/regula_falsi.cpp [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+#include<iostream>
+#include<cmath>
+// approximation error
+#define error 0.001
+using namespace std;
+
+// f(x) = x^3 -2*x -5
```
Comment (asha15) [inline Regula_Falsi_Method/regula_falsi.cpp]: take this commend before the return statement

File: Regula_Falsi_Method/regula_falsi.cpp [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+float polynomial(float x)    
+{
+    return ( ( x * x * x ) - 3 * x - 5);
+}
+
+int main()
+{
+    float x1,x2,x3,f1,f2,f3 = INT_MAX;
```
Comment (asha15) [inline Regula_Falsi_Method/regula_falsi.cpp]: keep space after comma

File: Regula_Falsi_Method/regula_falsi.cpp [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    int i = 1;
+    cout << "\nFor the interval (x1,x2) where the root is to be searched- \n";
```
Comment (asha15) [inline Regula_Falsi_Method/regula_falsi.cpp]: keep space after comma

File: Regula_Falsi_Method/regula_falsi.cpp [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    cout << "Enter x1 and x2 : " << endl;
+    cin >> x1 >> x2;
```
Comment (asha15) [inline Regula_Falsi_Method/regula_falsi.cpp]: keep a line space

File: Regula_Falsi_Method/regula_falsi.cpp [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    while(abs(f3) > error)
+    {   f1 = polynomial(x1);
```
Comment (asha15) [inline Regula_Falsi_Method/regula_falsi.cpp]: take the f1 into the next line

File: Regula_Falsi_Method/regula_falsi.cpp [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+        f2 = polynomial(x2);
+        x3 =x1 - (x2 - x1) / (f2 - f1) * f1 ;
+        f3=polynomial(x3);
```
Comment (asha15) [inline Regula_Falsi_Method/regula_falsi.cpp]: keep space around equal

File: Regula_Falsi_Method/regula_falsi.cpp [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+        if (f1 * f3 > 0)
```
Comment (asha15) [inline Regula_Falsi_Method/regula_falsi.cpp]: keep a line space before this

File: Regula_Falsi_Method/regula_falsi.cpp [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+            x1 = x3;
+        }
+        cout << "\nThe no. of iteration : " << i << " root is : " << x3 << endl;
+        i++;
+    }
+    cout << "\nThe approximated root of the polynomial that lies in the given interval is : " << x3 << endl;
+    return 0;
+}
+
+
```
Comment (asha15) [inline Regula_Falsi_Method/regula_falsi.cpp]: remove this lines

File: Regula_Falsi_Method/regula_falsi.cpp [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+
+/*
+OUTPUT==================================================================================================================
```
Comment (asha15) [inline Regula_Falsi_Method/regula_falsi.cpp]: remove the equal signs

File: Regula_Falsi_Method/regula_falsi.cpp (hunk 1 above)
Comment (Nikhil-Patro) [reply Regula_Falsi_Method/regula_falsi.cpp]: U mean C++ right?

**Comment by Nikhil-Patro** [comment]
All changes addressed and all commits squashed into one.Please review @asha15

**Comment by asha15** [comment]
Good job :)

**Pull request merged**